bitswap/provider.go

```diff
@@ -325,13 +339,15 @@ func (provider *Provider) handleResponses() {
 	}
 }
 
-func (provider *Provider) handleRetrievals() {
-	ctx := context.Background()
-
-	for {
+func (provider *Provider) handleRetrievals(ctx context.Context) {
+	for ctx.Err() == nil {
 		peerID, tasks, _ := provider.retrievalQueue.PopTasks(1)
```

Comment:
now that retreival is synchronous, this appears to limit things to one retrieval per worker queue, no?

Reply:
yes, default is 8 workers, but I hadn't seen a reason to increase this (yet) because it appears the pipe of incoming requests is so small; but maybe I'm not seeing it right